### ISSUE TITLE
[feat] Place replied-to indicator above message and username

### ISSUE THREAD

**Issue by ShadowJonathan** [ajbura/cinny#406]
### Describe the problem

Currently I find the way replies are marked in cinny pretty awkward, as replies do not really read comfortably to me;

![image](https://user-images.githubusercontent.com/22740616/159268280-78976f12-e8db-497a-afd9-a3c49b664326.png)

This to me reads as;
- Message
  - Author: Foxpy
  - Inner:
    - Reply-to:
      - Author: Silicius
      - Text: `image.jpeg`
    - Text: lol

### Describe the solution you'd like

Place the reply indicator above the message column, like so;

![image](https://user-images.githubusercontent.com/22740616/159269570-0d7ce419-1738-4177-a983-31d469c68c3d.png)

This to me would read as;
- Context: Reply-to: (Silicius, `image.jpeg`)
- Message:
  - Author: foxpy
  - Text: lol

Which parses much better.

It would also help to place any "external" "before" context *outside* of the message like this, as that'd help conceiving the message.

### Additional context

Discord does this pattern, and I think it works well.

Element has a pattern of "inlining" the response, of which cinny takes after here;

![image](https://user-images.githubusercontent.com/22740616/159269315-80ae6a68-5ab1-48c3-b138-63e32e43cb1f.png)

However, this pattern works "better" as this is about a blockquote indent, which has traditionally been used to quote specific bits of a message, which Element inherits from its historical design decisions.

However, here, cinny uses a "referral" arrow, one which works better by pointing at the "whole" message, instead of just the contents. Using it in place of the blockquote indicator confuses what it is supposed to be, and by extension, is taxing to parse.

In short, its a mix of design pointers which don't work well together, thus this recommendation.

**Comment by ShadowJonathan** [comment]
I discovered a bit of nuance; This unconditional pattern doesn't work if there's replies "inline" in a large chain of text, so this'd either have to apply to only the first message in a chain, or it'd have to break the chain, and show the author and username again (this is what discord also does)

**Comment by clmnin** [comment]
I think the current approach is better?
Look at this image
![replies](https://user-images.githubusercontent.com/17064666/160147285-bd72cd48-b75d-4462-891c-d87cb898a833.png)
